firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -58,9 +76,29 @@ public void logArmActive(@NonNull String key, @NonNull ConfigContainer configCon
       return;
     }
 
-    Bundle params = new Bundle();
-    params.putString(ARM_KEY, metadata.optString(PERSONALIZATION_ID));
-    params.putString(ARM_VALUE, values.optString(key));
-    analyticsConnector.logEvent(ANALYTICS_ORIGIN_PERSONALIZATION, ANALYTICS_PULL_EVENT, params);
+    String choiceId = metadata.optString(CHOICE_ID);
+    if (choiceId.isEmpty()) {
+      return;
+    }
+
+    synchronized (armsCache) {
+      if (choiceId.equals(armsCache.get(key))) {
+        return;
+      }
+      armsCache.put(key, choiceId);
+    }
+
+    Bundle logParams = new Bundle();
+    logParams.putString(ARM_KEY, key);
```

Comment:
Minor: there are a lot of things called "key" in this file. Since the `key` arg passed to the `logArmActive` fn is actually the RC param, renaming it to something like `rcParameter` would be more self-descriptive.